### PR TITLE
feat(vehicle): open source the anbox VHAL HIDL interface

### PR DESCRIPTION
**Pull request by adglkh**
To enable the third party VHAL implementation integrating into Anbox
Cloud, and view/modify property values from Anbox Cloud dashboard,
we now open source the Anbox specific VHAL HIDL interface, which
allows Android system developers to extend or customize HAL
implementations to meet specific hardware requirements, simplify
integration with Anbox runtime.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# Vendor Canonical Interfaces
```
Comment (morphis) [inline README.md]: The title should definitely include Anbox

File: README.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+## Description
+
+This repository contains the interface definitions for Anbox specific hardware abstraction layer (HAL), including HIDL specifications, along with reference implementations for some interfaces, such as VHAL. This allows Android system developers to extend or customize HAL implementations to meet specific hardware requirements, simplify integration with Anbox runtime.
```
Comment (morphis) [inline README.md]:
```suggestion
This repository contains the interface definitions for Anbox specific hardware abstraction layer (HAL), including HIDL specifications, along with reference implementations for some interfaces, such as VHAL. This allows Android system developers to extend or customize HAL implementations to meet specific hardware requirements, simplify integration with the Anbox runtime.
```

File: README.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+2. Ensure that your code adheres to the established [coding standards and practices](https://ubuntu.com/community/ethos/code-of-conduct).
+3. Test your changes to ensure that automated tests pass with the modifications you made.
+4. Provide relevant documentation updates if needed.
+5. Sign the [contributor agreement](https://ubuntu.com/legal/contributors), submit a pull request, outlining the purpose and scope of your changes.
+
+Our team will review your contribution and collaborate with you to integrate it into the SDK.
+
+## Bug report
+
+To report issues, please submit a [bug](https://bugs.launchpad.net/anbox-cloud/+filebug) to Anbox Cloud project on launchpad.
```
Comment (morphis) [inline README.md]:
```suggestion
To report issues, please submit a [bug](https://bugs.launchpad.net/anbox-cloud/+filebug) to the Anbox Cloud project on Launchpad.
```

File: README.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+## License
+
+The Vendor Canonical Interfaces project is licensed under the Apache License 2.0. For more details, refer to the [LICENSE](LICENSE) file for details.
```
Comment (morphis) [inline README.md]: Needs to be adjusted according to the updated title.

File: vehicle/1.0/Android.bp [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package {
+    // See: http://go/android-license-faq
+    // A large-scale-change added 'default_applicable_licenses' to import
+    // all of the 'license_kinds' from "hardware_interfaces_license"
+    // to get the below license kinds:
+    //   SPDX-license-identifier-Apache-2.0
```
Comment (morphis) [inline vehicle/1.0/Android.bp]: The above is no longer true and can be dropped.

File: vehicle/1.0/IVehicle.hal [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+interface IVehicle {
+  /**
+   * Get a vehicle property value.
+   *
+   * For VehiclePropertyChangeMode::STATIC properties, this method must always
+   * return the same value always.
+   * For VehiclePropertyChangeMode::ON_CHANGE properties, it must return the
+   * latest available value.
+   *
+   * Some properties like RADIO_PRESET requires to pass additional data in
```
Comment (morphis) [inline vehicle/1.0/IVehicle.hal]:
```suggestion
   * Some properties like RADIO_PRESET require to pass additional data in
```

File: vehicle/Android.bp [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package {
+    default_applicable_licenses: ["canonical_hardware_interfaces_license"],
+}
+
+// Added automatically by a large-scale-change
+// See: http://go/android-license-faq
```
Comment (morphis) [inline vehicle/Android.bp]: See above.

**Review by morphis — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (adglkh) [reply README.md]: +1, It makes sense to me

File: vehicle/1.0/Android.bp (hunk 5 above)
Comment (adglkh) [reply vehicle/1.0/Android.bp]: +1

File: README.md [hunk 8]
```diff
@@ -0,0 +1,36 @@
+# Canonical Anbox Hardware Interfaces
```
Comment (morphis) [inline README.md]:
```suggestion
# Android HAL interfaces for Canonical Anbox Cloud
```

File: README.md [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+
+Our team will review your contribution and collaborate with you to integrate it into the SDK.
+
+## Bug report
+
+To report issues, please submit a [bug](https://bugs.launchpad.net/anbox-cloud/+filebug) to Anbox Cloud project on Launchpad.
+
+## License
+
+The Canonical Anbox Hardware Interfaces project is licensed under the Apache License 2.0. For more details, refer to the [LICENSE](LICENSE) file for details.
```
Comment (morphis) [inline README.md]:
```suggestion
The content of this repository is licensed under the terms of the Apache 2.0 license. For more details, refer to the [LICENSE](LICENSE) file for details.
```

**Review by morphis — CHANGES_REQUESTED** [review]

**Review by morphis — APPROVED** [review]
LGTM

File: README.md [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+
+Our team will review your contribution and collaborate with you to integrate it into the SDK.
+
+## Bug report
+
+To report issues, please submit a [bug](https://bugs.launchpad.net/anbox-cloud/+filebug) to Anbox Cloud project on Launchpad.
+
+## License
+
+The content of this repository is licensed under the terms of the Apache 2.0 license. For more details, refer to the [LICENSE](LICENSE) file for details.
```
Comment (keirthana) [inline README.md:36]: Is the license file linked correctly? Just checking once as it seems to have no path.

**Review by keirthana — APPROVED** [review]
Overall looks good, just one comment to double-check the license file link.

File: README.md (hunk 10 above)
Comment (adglkh) [reply README.md:36]: It points to the LICENSE file in the main branch, E.g.
https://github.com/canonical/vendor_canonical_interfaces/blob/main/LICENSE

**Pull request merged**